pkg/backup/abs_backend_test.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2017 The etcd-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backup
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/storage"
+	"github.com/coreos/etcd-operator/pkg/backup/abs"
+)
+
+const (
+	absIntegrationTestEnvVar = "RUN_ABS_INTEGRATION_TEST"
+	absIntegrationTestNotSet = "skipping ABS integration test due to RUN_ABS_INTEGRATION_TEST not set"
+)
+
+var (
+	accountName    = storage.StorageEmulatorAccountName
+	accountKey     = storage.StorageEmulatorAccountKey
+	DefaultBaseURL = "http://127.0.0.1:10000"
+	container      = "testcontainer"
+	prefix         = "testprefix"
+	blobContents   = "ignore"
+)
+
+func TestABSBackendContainerDoesNotExist(t *testing.T) {
+	if os.Getenv(absIntegrationTestEnvVar) != "true" {
+		t.Skip(absIntegrationTestNotSet)
+	}
+
+	storageClient, err := storage.NewClient(accountName, accountKey, DefaultBaseURL, "", false)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = abs.NewFromClient(container, prefix, &storageClient)
+	if err == nil {
+		t.Fatal(err)
+	}
+	if err.Error() != "container testcontainer does not exist" {
+		t.Fatal(err)
+	}
+}
+
+func TestABSBackendGetLatest(t *testing.T) {
+	if os.Getenv(absIntegrationTestEnvVar) != "true" {
+		t.Skip(absIntegrationTestNotSet)
+	}
+
+	storageClient, err := storage.NewClient(accountName, accountKey, DefaultBaseURL, "", false)
+	if err != nil {
+		t.Fatal(err)
+	}
+	blobServiceClient := storageClient.GetBlobService()
+
+	// Create container
+	cnt := blobServiceClient.GetContainerReference(container)
+	options := storage.CreateContainerOptions{
+		Access: storage.ContainerAccessTypePrivate,
+	}
+	_, err = cnt.CreateIfNotExists(&options)
+	if err != nil {
+		if accountName == storage.StorageEmulatorAccountName {
```

Comment:
isn't `if accountName == storage.StorageEmulatorAccountName` always true?
what's this check for?

Reply:
Good call; not needed.